### PR TITLE
Likhith/86119/display markets under correct tabs

### PR DESCRIPTION
**Pull request by likhith-deriv**

### CODE REVIEW

File: src/components/index.js [hunk 1]
```diff
@@ -1,12 +1,17 @@
-import Watchlist from "./watchlist";
-import SVGWrapper from "./svg-wrapper";
-import NavBar from "./nav";
+import { Tab, Tabs } from "./tabs";
+
 import AccountSwitcher from "./account-switcher";
+import Button from "./button";
+import DataTable from "./data-table";
+import DisplayChangePercent from "./displasy-change-percent";
```
Comment (farrah-deriv) [inline src/components/index.js]:
```suggestion
import DisplayChangePercent from "./display-change-percent";
```

File: src/components/loader.jsx [hunk 2]
```diff
@@ -1,8 +1,29 @@
-import Styles from "../styles/loader.module.scss";
+import { Match, Switch } from "solid-js";
+
```
Comment (farrah-deriv) [inline src/components/loader.jsx]: please remove this

File: src/components/market-list.jsx [hunk 3]
```diff
@@ -0,0 +1,308 @@
...
+} from "solid-js";
+import {
+  activeSymbols,
+  fetchMarketTick,
+  market_ticks,
+  setMarketTicks,
+} from "../stores";
+import { addDays, formatDate } from "../utils/format-value";
+import { forgetAll, sendRequest, wait } from "../utils/socket-base";
+
```
Comment (farrah-deriv) [inline src/components/market-list.jsx]: please remove this

File: src/components/market-list.jsx [hunk 4]
```diff
@@ -0,0 +1,308 @@
...
+  onMount(() => {
+    setActiveTab(0);
+    setIsMarketClosed(false);
+  });
+
+  createEffect(() => {
+    setAllMarkets(segregateMarkets(activeSymbols()));
+    if (active_tab() === 0) getWatchList();
+    console.log("All mkst: ", all_markets());
```
Comment (farrah-deriv) [inline src/components/market-list.jsx]: do we need this log here?

File: src/components/market-list.jsx [hunk 5]
```diff
@@ -0,0 +1,308 @@
...
+
+  const getSymbol = (target_symbol, trading_times) => {
+    let symbol;
+    const { markets } = trading_times;
+    for (let i = 0; i < markets.length; i++) {
+      const { submarkets } = markets[i];
+      for (let j = 0; j < submarkets.length; j++) {
+        const { symbols } = submarkets[j];
+        symbol = symbols.find((item) => item.symbol === target_symbol);
+        if (symbol !== undefined) return symbol;
```
Comment (farrah-deriv) [inline src/components/market-list.jsx]:
```suggestion
        if (symbol) return symbol;
```

File: src/components/data-table.jsx [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+              </For>
+            </tr>
+          </thead>
+        </Show>
+        <tbody class={classNames(styles["table-body"], props.table_body_class)}>
+          <For each={props.data}>
+            {(cell_value, index) => (
+              <tr>
+                <td
+                  onClick={() => console.log("TD clicked")}
```
Comment (farrah-deriv) [inline src/components/data-table.jsx]: do we need to log here?

File: src/components/index.js (hunk 1 above)
Comment (likhith-deriv) [reply src/components/index.js]: Ok

File: src/components/loader.jsx (hunk 2 above)
Comment (likhith-deriv) [reply src/components/loader.jsx]: Will remove the setting

File: src/components/market-list.jsx (hunk 3 above)
Comment (likhith-deriv) [reply src/components/market-list.jsx]: Will remove the setting

File: src/components/market-list.jsx (hunk 4 above)
Comment (likhith-deriv) [reply src/components/market-list.jsx]: No. Will remove it

File: src/components/market-list.jsx (hunk 5 above)
Comment (likhith-deriv) [reply src/components/market-list.jsx]: ok

File: src/components/data-table.jsx (hunk 6 above)
Comment (likhith-deriv) [reply src/components/data-table.jsx]: No

File: src/components/data-table.jsx [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+                    }
+                  </For>
+                </td>
+              </tr>
+            )}
+          </For>
+        </tbody>
+      </table>
+
+      <table
```
Comment (nijil-deriv) [inline src/components/data-table.jsx]: What do we need a 2nd table for?

Comment (likhith-deriv) [reply src/components/data-table.jsx]: Refactored the code

**Review by vinu-deriv — APPROVED** [review]

**Pull request merged**